src/overlays/fbdemos/ovl_fbdemo_triforce/z_fbdemo_triforce.c

```diff
@@ -15,35 +16,135 @@ void TransitionTriforce_SetType(void* thisx, s32 type);
 void TransitionTriforce_SetColor(void* thisx, u32 color);
 s32 TransitionTriforce_IsDone(void* thisx);
 
-#if 0
 const TransitionInit TransitionTriforce_InitVars = {
-    TransitionTriforce_Init,
-    TransitionTriforce_Destroy,
-    TransitionTriforce_Update,
-    TransitionTriforce_Draw,
-    TransitionTriforce_Start,
-    TransitionTriforce_SetType,
-    TransitionTriforce_SetColor,
-    NULL,
+    TransitionTriforce_Init,   TransitionTriforce_Destroy, TransitionTriforce_Update,   TransitionTriforce_Draw,
+    TransitionTriforce_Start,  TransitionTriforce_SetType, TransitionTriforce_SetColor, NULL,
     TransitionTriforce_IsDone,
 };
 
-#endif
+void TransitionTriforce_Start(void* thisx) {
+    TransitionTriforce* this = (TransitionTriforce*)thisx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/TransitionTriforce_Start.s")
+    switch (this->state) {
+        case 1:
+        case 2:
+            this->transPos = 1.0f;
+            return;
+    }
+    this->transPos = 0.03f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/TransitionTriforce_Init.s")
+void* TransitionTriforce_Init(void* thisx) {
+    TransitionTriforce* this = (TransitionTriforce*)thisx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/TransitionTriforce_Destroy.s")
+    bzero(this, sizeof(*this));
+    guOrtho(&this->projection, -160.0f, 160.0f, -120.0f, 120.0f, -1000.0f, 1000.0f, 1.0f);
+    this->transPos = 1.0f;
+    this->state = 2;
+    this->step = 0.015f;
+    this->fadeDirection = 1;
+    return this;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/TransitionTriforce_Update.s")
+void TransitionTriforce_Destroy(void* thisx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/TransitionTriforce_SetColor.s")
+void TransitionTriforce_Update(void* thisx, s32 updateRate) {
+    TransitionTriforce* this = (TransitionTriforce*)thisx;
+    f32 temp_f0;
+    s32 i;
```

Comment:
```suggestion
    s32 pad;
    s32 i;
```
And it can be removed if the pad isn't needed for stack purposes